src/cmd/troubleshoot/proxy.go

```diff
@@ -0,0 +1,129 @@
+package troubleshoot
+
+import (
+	"github.com/Dynatrace/dynatrace-operator/src/dtclient"
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects"
+	"github.com/go-logr/logr"
+	"golang.org/x/net/http/httpproxy"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func checkProxySettings(troubleshootCtx *troubleshootContext) error {
+	return checkProxySettingsWithLog(troubleshootCtx, newTroubleshootLogger("[proxy     ] "))
+}
+
+func checkProxySettingsWithLog(troubleshootCtx *troubleshootContext, logger logr.Logger) error {
+	log = logger
+
+	var proxyURL string
+	logNewTestf("Analyzing proxy settings ...")
+
+	proxySettingsAvailable := false
+	if troubleshootCtx.dynakube.HasProxy() {
+		proxySettingsAvailable = true
+		logInfof("Reminder: Proxy settings in the Dynakube do not apply to pulling of pod images. Please set your proxy on accordingly on node level.")
+		logWarningf("Proxy settings in the Dynakube are ignored for codeModules images due to technical limitations.")
+
+		var err error
+		proxyURL, err = getProxyURL(troubleshootCtx)
+		if err != nil {
+			logErrorf("Unexpected error when reading proxy settings from Dynakube: %v", err)
+			return nil
+		}
+	}
+
+	if checkEnvironmentProxySettings(proxyURL) {
+		proxySettingsAvailable = true
+	}
+
+	if !proxySettingsAvailable {
+		logOkf("No proxy settings found.")
+	}
+	return nil
+}
+
+func checkEnvironmentProxySettings(proxyURL string) bool {
+	envProxy := getEnvProxySettings()
+	if envProxy == nil {
+		return false
+	}
+
+	logInfof("Searching environment for proxy settings ...")
+	if envProxy.HTTPProxy != "" {
+		logWarningf("HTTP_PROXY is set in environment. This setting will be used by the operator for codeModule image pulls.")
+	}
+	if envProxy.HTTPSProxy != "" {
+		logWarningf("HTTPS_PROXY is set in environment. This setting will be used by the operator for codeModule image pulls.")
+	}
+	if envProxy.HTTPProxy != proxyURL && envProxy.HTTPSProxy != proxyURL {
+		logWarningf("Proxy settings in the Dynakube and proxy settings in the pod environment differ.")
+	}
+	return true
+}
+
+func getEnvProxySettings() *httpproxy.Config {
+	envProxy := httpproxy.FromEnvironment()
+	if envProxy.HTTPProxy != "" || envProxy.HTTPSProxy != "" {
+		return envProxy
+	}
+	return nil
+}
+
+func applyProxySettings(troubleshootCtx *troubleshootContext) error {
+	proxyURL, err := getProxyURL(troubleshootCtx)
+	if err != nil {
+		return err
+	}
+
+	if proxyURL != "" {
+		err := troubleshootCtx.SetTransportProxy(proxyURL)
+		if err != nil {
+			return errorWithMessagef(err, "error parsing proxy value")
+		}
+	}
```

Comment:
If's `true` branch is not covered with unit tests but it seems worth it.

Reply:
True, but this is used by dynakube checks and not by image pull checks. Has to be done in a separate PR. I do not want to  prolong this one any more than absolutely necessary by getting side tracked.